src/server/JasmineGraphInstanceService.cpp

```diff
@@ -3584,3 +3650,309 @@ map<long, map<long, unordered_set<long>>> calculateEgoNet(string graphID, string
+                                                                                    centralDB,
+                                                                                    localGraphMap, workerSockets,
+                                                                                    worldOnlyVertexCount);
+
+    map<long, unordered_set<long>> hMapFromVerticesWorldToLocalFLow = getEdgesWorldToLocal(graphID, partitionID,
+                                                                                           serverPort, graphVertexCount,
+                                                                                           localDB, centralDB,
+                                                                                           localGraphMap,
+                                                                                           workerSockets);
+
+    long entireGraphSize = atol(graphVertexCount.c_str());
+    int T = 2;
+    float mu = (damp / entireGraphSize);
+    int ITERATION_COUNT = 10;
+    unordered_map<float, float> resultTreeMap;
+    // calculating local pagerank
+    map<long, double> rankMap;
+
+    // add world to local authority scores to the ranks of the current partition
+    for (auto &hMapAuthorityFlowWorldToLocalIterator: hMapAuthorityFlowWorldToLocal) {
+        long vertex = hMapAuthorityFlowWorldToLocalIterator.first;
+        float worldToLocalRank = hMapAuthorityFlowWorldToLocalIterator.second;
+
+        auto rankMapItr = rankMap.find(vertex);
+        if (rankMapItr != rankMap.end()) {
+            double existingRank = rankMapItr->second;
+            double finalRank = existingRank + worldToLocalRank;
+            rankMapItr->second = finalRank;
+        } else {
+            rankMap.insert(std::make_pair(vertex, worldToLocalRank));
+        }
+    }
+
+    int count = 0;
+    while (count < ITERATION_COUNT) {
+
+        for (localGraphMapIterator = localGraphMap.begin();
+             localGraphMapIterator != localGraphMap.end(); ++localGraphMapIterator) {
+
+            long startVid = localGraphMapIterator->first;
+            unordered_set<long> endVidSet = localGraphMapIterator->second;
+            double existingParentRank = 1;
+
+            auto rankMapItr = rankMap.find(startVid);
+            if (rankMapItr != rankMap.end()) {
+                existingParentRank = rankMapItr->second;
+            } else {
+                rankMap.insert(std::make_pair(startVid, existingParentRank));
+            }
+
+            long degree = endVidSet.size();
+            double distributedRank = alpha * (existingParentRank / degree) + mu;
+
+            for (long itr: endVidSet) {
+                auto rankMapItr = rankMap.find(itr);
+
+                double existingChildRank = 0;
+                if (rankMapItr != rankMap.end()) {
+                    existingChildRank = rankMapItr->second;
+                    double finalRank = existingChildRank + distributedRank;
```

Comment:
This variable `double finalRank` can be declared just after the line `double existingChildRank = 0;`

Reply:
Fixed

Reply:
Ok